docs/docs/schema-customization.md

```diff
@@ -0,0 +1,591 @@
+# Customizing the GraphQL Schema
+
+One of Gatsby's main strengths is the ability to query data from a variety of
+sources in a uniform way with GraphQL. For this to work, a GraphQL Schema must
+be generated that defines the shape of the data.
+
+Gatsby is able to automatically infer a GraphQL Schema from your data, and in
+many cases this is really all you need. There are however situations when you
+either want to explicitly define the data shape, or add custom functionality to
+the query layer - this is what Gatsby's Schema Customization API provides.
+
+The following guide walks through a basic example to showcase the API.
+
```

Comment:
What do you think about adding a section describing the target audience for this guide, and any prerequisite knowledge? 

The target audience might be something like plugin authors, developers optimising builds for larger sites, or anyone interested in learning more about how Gatsby uses GraphQL. 

Required knowledge could be some understanding of using `gatsby-node.js` to customise your site's build, along with basic familiarity with GraphQL's types and schemas?